draft-jmiller-json-proof-algorithms.md

```diff
@@ -184,15 +184,163 @@ The Verifier uses only the disclosed payloads and generates or uses the included
+This JSON Proof Algorithm definition for BBS is based on the already released implementation and relies on the provided software API, a future definition with a different `alg` value will be created to succeed this version as the BBS standardization effort progresses.
 
-Proposed JWK `alg` value is "BBS" with a `crv` value of "Bls12381G2".
+This algorithm supports both selective disclosure and unlinkability, enabling the Prover to generate multiple proofs from one signed JWP without the verifier being able to correlate those proofs together.
 
-Proposed JWP `alg` value is "BBS-Bls12381G2".
+### BLS Curve
+
+The pairing friendly elliptic curve used for the BBS software implementation is part of the BLS family with an embedding degree of 12 over a 381-bit prime field.  For this JPA, only the group G2 is used.
+
+In the implementation the method used to generate the key pairs is `generateBls12381G2KeyPair()`.
+
+### Messages
+
+BBS is a multi-message scheme and operates on an array of individual messages for signing and proof generation.  Each message is a binary octet string, the BBS implementation uses a hash-to-curve method to map each message to a point.
+
+### Protected Header
+
+The UTF-8 octet string of the JWP Protected Header is the first message in the input array, it is always at index 0.
+
+### Payloads
+
+The octet strings of each payload are placed into the BBS message array following the protected header message.  For example, first payload is at index 1 of the array and the last payload is always the last message in the array.
+
+In future versions of this algorithm there will be additional methods defined for transforming a payload into a point such that additional Zero-Knowledge Proof types can be supported by the Prover such as range and membership predicates.
+
+### Signing
+
+The Signer's BLS12-381 G2 key pair is used to sign the completed message array input containing the octet strings of the Protected Header and every payload.  The result is a signature octet string that is used as the signed JWP Proof value.
+
+In the implementation, the method used to perform the signing is `blsSign({keyPair, [header, payload1, payload2, ...]})` and returns a binary signature value.
+
+### Proving
+
+The prover must decode the JWP header and payload values in order to generate the identical message array that the signer used.
+
+To generate a JWP for a verifier, the prover must use a cryptographic nonce that is provided by that verifier as input.  This nonce MUST be a 32 byte octet string that the verifier generated from a trusted  source.
+
+The prover also applies selective disclosure preferences by creating an array of indices of which messages in the input array are to be revealed to the verifier.  The revealed indices MUST include the value `0` so that the protected header message is always revealed to the verifier.
```

Comment:
Since this first definition is based on the existing implementation, the language here is defining how to map the inputs into that software.  The revealed indices array argument _must_ start with a `0` as the first entry in this case in order to use the software correctly for a JWP.